plugin/trino-mysql/src/main/java/io/trino/plugin/mysql/MySqlJdbcConfig.java

```diff
@@ -42,8 +41,11 @@ public boolean isUrlWithoutDatabase()
     {
         try {
             Driver driver = new Driver();
-            Properties properties = driver.parseURL(getConnectionUrl(), null);
-            return (properties == null) || (driver.database(properties) == null);
+            if (driver.acceptsURL(getConnectionUrl())) {
+                String path = ConnectionUrlParser.parseConnectionString(getConnectionUrl()).getPath();
+                return path == null || path.strip().equals("");
+            }
+            return false;
```

Comment:
Should we throw here instead? It will already be handled by the `isUrlValid` validation.

Reply:
Or remove the `if` and allow the `ConnectionUrlParser.parseConnectionString` to throw.

Reply:
@hashhar no, the more detailed validation should "succeed" when the more coarse is known to fail.
otherwise you can multiple validation errors, unnecessarily.